modules/shared/src/main/scala/org/tessellation/schema/swap.scala

```diff
@@ -0,0 +1,78 @@
+package org.tessellation.schema
+
+import org.tessellation.schema.address.Address
+import org.tessellation.security.hash.Hash
+
+import derevo.cats.{order, show}
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import eu.timepit.refined.types.numeric.PosLong
+import io.estatico.newtype.macros.newtype
+
+object swap {
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class SwapAmount(value: PosLong)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class SwapSalt(value: Long)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class AllowSpendFee(value: PosLong)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class CurrencyId(value: Address)
+
+  @derive(decoder, encoder, order, show)
+  @newtype
+  case class AllowSpendReference(value: Hash) // exact type TBD (e.g. lastRef?)
+
+  @derive(decoder, encoder, order, show)
+  case class AllowSpend(
+    source: Address,
+    destination: Address,
+    currency: CurrencyId,
+    amount: SwapAmount,
+    fee: AllowSpendFee,
+    parent: AllowSpendReference,
+    salt: SwapSalt,
```

Comment:
is the `salt` required?

Reply:
I think this was just a convention following the original transaction schema, I don't think it's really required there either due to unique transaction references. Not sure the original reason behind that.

Reply:
removed